pkg/executor/test/simpletest/simple_test.go

```diff
@@ -230,6 +230,67 @@ func TestRole(t *testing.T) {
 	tk.MustExec("SET ROLE NONE")
 }
 
+func TestMaxUserConnections(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+
+	// test global variables max_user_connections.
+	result := tk.MustQuery(`show variables like 'max_user_connections'`)
+	result.Check(testkit.Rows("max_user_connections 0"))
+	tk.MustExec(`set global max_user_connections = 3;`)
+	tk.MustQuery(`show variables like 'max_user_connections'`).Check(testkit.Rows("max_user_connections 3"))
+	// if the value < 0, set 0 to max_user_connections.
+	tk.MustExec(`set global max_user_connections = -1;`)
+	tk.MustQuery(`show variables like 'max_user_connections'`).Check(testkit.Rows("max_user_connections 0"))
+	// if the value > 100000, set 100000 to max_user_connections.
+	tk.MustExec(`set global max_user_connections = 100001;`)
+	tk.MustQuery(`show variables like 'max_user_connections'`).Check(testkit.Rows("max_user_connections 100000"))
+	tk.MustExec(`set global max_user_connections = 0;`)
+	tk.MustQuery(`show variables like 'max_user_connections'`).Check(testkit.Rows("max_user_connections 0"))
+
+	// create user with the default max_user_connections 0
+	createUserSQL := `CREATE USER 'test'@'localhost';`
+	tk.MustExec(createUserSQL)
+	result = tk.MustQuery(`select user, max_user_connections from mysql.user`)
+	result.Check(testkit.Rows("root 0", "test 0"))
+
+	// create user with max_user_connections 3
+	createUserSQL = `CREATE USER 'test1'@'localhost' WITH MAX_USER_CONNECTIONS 3;`
+	tk.MustExec(createUserSQL)
+	result = tk.MustQuery(`select user, max_user_connections from mysql.user WHERE User="test1"`)
+	result.Check(testkit.Rows("test1 3"))
+
+	// test alter user with MAX_USER_CONNECTIONS
```

Comment:
Will this MAX_USER_CONNECTIONS changes impact the current user who did not set this value in his definition？

Reply:
If the operation of 'create user' doesn't contain the option of 'WITH MAX_USE_CONNECTIONS', the user has the MAX_USER_CONNECTIONS with default 0.
The operation of 'alter user ... with MAX_USER_CONNECTIONS' can make impact on this scenario and can valid.